### PR TITLE
Adding Dockerfiles

### PR DESCRIPTION
**Pull request by ruffsl**
We've been making some Dockerfiles for using this project and thought I'd might share them. You may like to use them as well to enable a CI setup like there exists for [MoveIt!](https://github.com/ros-planning/moveit/tree/kinetic-devel/.docker),  or for testing build from PRs and giving users a pre built examples to tryout.

### CODE REVIEW

File: .docker/Makefile [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+	@echo ""
+	@echo "   1. make build_cpu        - build cpu image"
+	# @echo "   1. make build_gpu        - build gpu image"
+	# @echo "   1. make pull             - pull all images"
+	# @echo "   1. make clean            - remove all images"
+	@echo ""
+
+build_cpu:
+		cd .. && \
+		docker build --tag=spencer/people_tracking:cpu --file=docker/cpu/Dockerfile .
```
Comment (tlind) [inline .docker/Makefile]: I think the file path is missing the leading dot (should be `.docker/cpu/Dockerfile`), right?

**Comment by ruffsl** [comment]
Ok, I've changed the layout to be distro orientated instead. If you want to add gpu targeted images later, perhaps that would best be under a different repo namespace. Perhaps you'd like to create a dockerhub organisation profile for the spencer project, then add an automated build repo for the tags/Dockerfiles here?

**Comment by tlind** [comment]
Great, thanks! I want to get #36 solved first, and then I will merge this PR. Regarding the dockerhub automated build, I'm no Docker expert, but I'll have a look at it when I find the time.

**Comment by ruffsl** [comment]
If you need more help with dockerhub, you can ping me anytime. I helped the MoveIt! project with that not to long ago.

**Pull request merged**